### PR TITLE
[13.0][MIG] shopinvader_product_attribute_set

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Standard migration:

- [x] Backport from 14.0
- [x] Fix Native field export
- [x] Add test for Native field

Depends on:
- [x] https://github.com/OCA/odoo-pim/pull/101

CC @ForgeFlow

### CODE REVIEW

**Comment by HviorForgeFlow** [comment]
Any suggestions about how to fix this? https://github.com/shopinvader/odoo-shopinvader/pull/1096/commits/760c1b74a1f8003e1d90f369705b98631dc889c9

Thanks!

File: shopinvader_product_attribute_set/models/shopinvader_variant.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return self._get_attribute_value(attr.name, one=True)
+        elif attr.attribute_type == "multiselect":
+            return self._get_attribute_value(attr.name)
+        elif string_mode and attr.attribute_type == "boolean":
+            return self[attr.name] and "true" or "false"
+        elif string_mode or attr.attribute_type in ("char", "text"):
+            return "%s" % (self[attr.name] or "")
+        elif attr.nature == "native" and attr.attribute_type == "float":
+            # TODO: Match digit precision something like:
+            # float_round(self[attr.name], precision_digits=attr.field_id.get_digits())
```
Comment (MiquelRForgeFlow) [inline shopinvader_product_attribute_set/models/shopinvader_variant.py]:
```suggestion
            float_round(self[attr.name], precision_digits=self.env[attr.field_id.model]._fields[attr.field_id.name].get_digits(self.env)[1])
```

**Comment by HviorForgeFlow** [comment]
@rousseldenis @sebastienbeau @Cedric-Pigeon can we move it forward? Thanks!

**Review by rousseldenis — APPROVED** [review]
LGTM

File: shopinvader_product_attribute_set/models/product_filter.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+from odoo import fields, models
+
+
+class ProductFilter(models.Model):
+    _inherit = "product.filter"
+
+    based_on = fields.Selection(
+        selection_add=[("product_attribute", "Product Attribute")],
+        ondelete="cascade",
```
Comment (acsonefho) [inline shopinvader_product_attribute_set/models/product_filter.py]: `ondelete="cascade"`? Why?

**Review by acsonefho — CHANGES_REQUESTED** [review]

File: shopinvader_product_attribute_set/models/product_filter.py (hunk 2 above)
Comment (acsonefho) [reply shopinvader_product_attribute_set/models/product_filter.py:14]: Ok that's correct.

**Review by acsonefho — APPROVED** [review]

**Review by acsonefho — CHANGES_REQUESTED** [review]
`ondelete="cascade"` doesn't exist in v13.0 for `Selection` fields.

**Comment by HviorForgeFlow** [comment]
> `ondelete="cascade"` doesn't exist in v13.0 for `Selection` fields.

ondelete removed from the selection field

**Review by PierrickBrun — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by HviorForgeFlow** [comment]
@rousseldenis @PierrickBrun can this be merged? Used in production for a while

**Pull request merged**